onnxconverter_common/auto_mixed_precision_model_path.py

```diff
@@ -0,0 +1,247 @@
+
+    model_32 = onnx.load(source_model_path)
+    save_model(model_32, tmp_model32_path, location=tmp_model32_tensor_name)
+
+    print("infer_shape_path for", tmp_model32_path, tmp_model32_tensor_name)
+    shape_inference.infer_shapes_path(tmp_model32_path)
+    model_32 = onnx.load(tmp_model32_path)
+
+    output_32 = inference(tmp_model32_path, input_feed, providers=providers)
+
+    kwargs["res1"] = output_32
+    kwargs["res2"] = output_32
+    if not _validate_result(**kwargs):
+        raise ValueError("validation failed for fp32 model")
+
+    return model_32, output_32
+
+
+def _validate_result(**kwargs):
+    validate_fn = kwargs.get("validate_fn")
+    rtol = kwargs.get("rtol")
+    res1 = kwargs.get("res1")
+    res2 = kwargs.get("res2")
+
+    if validate_fn is not None and not validate_fn(res1, res2):
+        return False
+    if rtol is not None:
+        for r1, r2 in zip(res1, res2):
+            if not np.allclose(r1, r2, rtol):
+                return False
+    return True
+
+
+def _convert_and_check_inference_result(**kwargs):
+    model_32 = kwargs.get("model_32")
+    keep_io_types = kwargs.get("keep_io_types")
+    is_final_model = kwargs.get("is_final_model")
+    target_model_path = kwargs.get("target_model_path")
+    node_block_list = kwargs.get("node_block_list")
+    input_feed = kwargs.get("input_feed")
+    providers = kwargs.get("providers")
+    tmp_model32_tensor_name = kwargs.get("tmp_model32_tensor_name")
+
+    _print_node_block_list(node_block_list)
+    # convert to fp16
+    model_16 = float16.convert_float_to_float16(
+        copy.deepcopy(model_32), node_block_list=node_block_list,
+        keep_io_types=keep_io_types, disable_shape_infer=True)
+    # need to save model to model_path here.....
+    if not is_final_model:
+        location = tmp_model32_tensor_name  # using temporary file name
+    else:
+        location = kwargs.get("location")  # using the speficified external data file name
+    save_model(model_16, target_model_path, location=location)
+    # inference
+    output_16 = inference(target_model_path, input_feed, providers=providers)
+    kwargs["res2"] = output_16
+    result = _validate_result(**kwargs)
+    print("validate result = ", result)
+    return result
+
+
+def inference(model_path, input_feed, providers=None):
+    sess = ort.InferenceSession(model_path, None, providers=providers)
+    output = sess.run(None, input_feed)
+    return output
+
+
+def save_model(model, model_path, location=None):
+    onnx.save_model(model, model_path, save_as_external_data=True, location=location)
+
+
+def _print_node_block_list(node_block_list, max_len=128):
+    print("node block list =")
+    if (len(node_block_list) < max_len):
+        print(node_block_list)
+    else:
+        tmp_list = node_block_list[0:64] + ['......'] + node_block_list[-64:]
+        print(tmp_list)
+
+
+def _clean_output_folder(**kwargs):
+    tmp_model32_path = kwargs.get("tmp_model32_path")
```

Comment:
Should we add a try block so that the exception here won't confuse users that the overall conversion might fail?